### PR TITLE
[CALCITE-2819] Add version of LoptOptimizeJoinRule that uses the first ordering it finds

### PR DESCRIPTION
**Pull request by siddharthteotia**

### CODE REVIEW

**Comment by siddharthteotia** [comment]
ping?

cc @laurentgo , @vladimirdolzhenko

File: core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java [hunk 1]
```diff
@@ -4292,6 +4293,20 @@ private Sql checkSubQuery(String sql) {
     checkPlanning(program, sql);
   }
 
+  @Test public void testConvertMultiJoinRuleLoptOptimizeJoinRuleNonOptimized() throws Exception {
```
Comment (vlsi) [inline core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java]: Does this really validate that the rule was executed only once?
It does not look so. At least from the brief analysis of the test code.

**Comment by vlsi** [comment]
@siddharthteotia , this PR adds extremely specific feature, and I don't see how the behavior is tested. Do you have a test that validates "only one"?

I think we could go with: "generate at most N permutations" or "spend no more than X milliseconds" rule parameters.

What do you think?

**Comment by siddharthteotia** [comment]
> @siddharthteotia , this PR adds extremely specific feature, and I don't see how the behavior is tested. Do you have a test that validates "only one"?
> 
> I think we could go with: "generate at most N permutations" or "spend no more than X milliseconds" rule parameters.
> 
> What do you think?

@vlsi , let me revisit the unit test and see if it can be changed/improved. I am a little confused with "generate at most N permutation" part. Are you referring to some piece of code that I can leverage to test this?

**Comment by vlsi** [comment]
>"generate at most N permutation" part. Are you referring to some piece of code that I can leverage to test this?

No.

I question the business purpose of the feature you add.
Why do you need "first ordering it finds"?

**Comment by siddharthteotia** [comment]
@vlsi ... we use this initially only to get some random ordering (not possibly the best ordering) to convert multi-join to join nodes.. There is probably no other way in Calcite to do that so added this unoptimized instance where we just return after getting the first ordering. So the test case also aims to verify if the result is the ordering by considering first join factor as the first factor.

**Comment by vlsi** [comment]
Where do you get MultiJoins in the first place?

**Comment by zabetak** [comment]
According to the JIRA it seems that we do not want to go towards this direction so I am closing down this PR. We can reopen it if needed.

**Pull request closed without merging**